app/src/org/commcare/tasks/CleanRawMedia.kt

```diff
@@ -0,0 +1,30 @@
+package org.commcare.tasks
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import org.commcare.CommCareApplication
+import org.commcare.android.database.user.models.FormRecord
+import org.commcare.utils.CrashUtil
+import java.io.File
+import java.lang.Exception
+
+class CleanRawMedia(appContext: Context, workerParams: WorkerParameters)
+    : Worker(appContext, workerParams) {
+
+    override fun doWork(): Result {
+        try {
+            val storage = CommCareApplication.instance().getUserStorage(FormRecord::class.java)
+            for (formRecord in storage) {
+                File(formRecord.filePath).parent?.let {
+                    val rawDirPath = "$it/raw"
+                    File(rawDirPath).deleteRecursively()
+                }
+            }
+        } catch (e: Exception) {
```

Comment:
Do we want to fail the loop for one failed record? Given what this task is doing I'd guess we should report each exception (And return failure for the task), but proceed with the next record. Otherwise one bad record will prevent cleaning up the remaining ones

Reply:
Yeah totally agree. Will add this.